test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -3810,6 +3810,45 @@ public void PackCommand_BuildOutput_DoesNotContainForSpecificFramework(string fr
             }
         }
 
+        [PlatformFact(Platform.Windows)]
+        public void PackCommand_BuildOutput_DoesNotContainDefaultExtensions()
+        {
+            // Arrange
+            using (var testDirectory = msbuildFixture.CreateTestDirectory())
+            {
+                var projectName = "ClassLibrary1";
+                var workingDirectory = Path.Combine(testDirectory, projectName);
+                var projectFile = Path.Combine(workingDirectory, $"{projectName}.csproj");
+                msbuildFixture.CreateDotnetNewProject(testDirectory.Path, projectName);
+
+                using (var stream = new FileStream(projectFile, FileMode.Open, FileAccess.ReadWrite))
+                {
+                    var xml = XDocument.Load(stream);
+                    ProjectFileUtils.SetTargetFrameworkForProject(xml, "TargetFramework", "net5.0");
+                    ProjectFileUtils.AddProperty(xml, "DefaultAllowedOutputExtensionsInPackageBuildOutputFolder", ".dll");
+                    ProjectFileUtils.AddProperty(xml, "GenerateDocumentationFile", "true");
+                    ProjectFileUtils.WriteXmlToFile(xml, stream);
+                }
+
+                msbuildFixture.RestoreProject(workingDirectory, projectName, string.Empty);
+
+                // Act
+                msbuildFixture.PackProject(workingDirectory, projectName, $"-o {workingDirectory}");
+
+                var nupkgPath = Path.Combine(workingDirectory, $"{projectName}.1.0.0.nupkg");
+                var nuspecPath = Path.Combine(workingDirectory, "obj", $"{projectName}.1.0.0.nuspec");
+                Assert.True(File.Exists(nupkgPath), "The output .nupkg is not in the expected place");
+                Assert.True(File.Exists(nuspecPath), "The intermediate nuspec file is not in the expected place");
+
+                // Assert
+                using (var nupkgReader = new PackageArchiveReader(nupkgPath))
+                {
+                    var buildItems = nupkgReader.GetBuildItems().ToList();
```

Comment:
Does this pass?

This sohuld only capture build files?

I'd recommend looking at the exact file list, as then you are not at the mercy of the implementation of the GetBuildItems method.

Reply:
i tested the fix standalone.
You are right, this should fail. i was not able to run the test using the local copy of  targets file, it end up using the my machine copy ? how can i fix that and run the test

Reply:
The tests should be copying the internal targets.

You can debug it here: https://github.com/NuGet/NuGet.Client/blob/7e3b0e445edecbdb6c46895ee4375cf834f45d87/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs#L570-L571

Reply:
> C:\git\NuGet.Client\.test\work\cd78165c\b5afa28a\sdk\5.0.200-servicing.21120.4\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(279,15): error MSB4064: The "Readme" parameter is not supported by the "PackTask" task. Verify the parameter exists on the task, and it is a settable public instance property. [C:\git\NuGet.Client\.test\work\c783b310\bbfe3ac5\ClassLibrary1\ClassLibrary1.csproj]
C:\git\NuGet.Client\.test\work\cd78165c\b5afa28a\sdk\5.0.200-servicing.21120.4\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(222,5): error MSB4063: The "PackTask" task could not be initialized with its input parameters.  [C:\git\NuGet.Client\.test\work\c783b310\bbfe3ac5\ClassLibrary1\ClassLibrary1.csproj]

i was able to run the tests but i get this as an error. i verified the file being copied is correct. @nkolev92 can u help me with this ?

Reply:
> The "Readme" parameter is not supported by the "PackTask" task

hmm, this suggests to me that targets and dlls are somehow out of sync. 

Maybe try clearing the artifacts folder, build from the commandline by running `msbuild /restore NuGet.sln /v:m`, and then running the test again. 
If not of that works, try pinging me offline.

Reply:
@nkolev92 that worked thanks. I modified to check for the specific file and test fails before and passes after the change.